eclass/freebsd.eclass

```diff
@@ -153,9 +162,23 @@ freebsd_src_unpack() {
 			[[ -e "${WORKDIR}"/share/mk ]] && rm -rf "${WORKDIR}"/share/mk/*.mk
 		fi
 	else
-		for f in ${A} ; do
-			[[ ${f} == *.tar.* ]] && unpack ${f}
-		done
+		if version_is_at_least 10.0 ${RV} ; then
+			local tarball="freebsd-src-${MY_PV}.tar.xz"
+			local topdir="usr/src/"
+			local extractlist
+			for i in ${EXTRACTONLY} ; do
+				extractlist+=" ${topdir}${i}"
+			done
+			ebegin "Unpacking parts of ${tarball} to ${WORKDIR}"
+			pushd "${WORKDIR}" > /dev/null
+			# https://bugs.gentoo.org/show_bug.cgi?id=550734
+			xzdec -c "${DISTDIR}/${tarball}" | tar -xpf - --strip-components=2 ${extractlist} 2> /dev/null || die "tar extract command failed"
+			popd > /dev/null
```

Comment:
||die

Also, you can alternatively use:

    cd "${WORKDIR}" || die
    ...
    cd - || die